code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -273,35 +276,57 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
+		if(is_type_in_list(target_move, GLOB.ventcrawl_machinery))
+			if(target_move.can_crawl_through())
+				user.remove_ventcrawl()
+				user.forceMove(target_move.loc) //handles entering and so on
+				user.visible_message("<span class='notice'>You hear something squeezing through the ducts.</span>", \
+				"<span class='notice'>You climb out the ventilation system.</span>")
+			else if(istype(user, /mob/living/silicon/robot/drone))
+				to_chat(user, "<span class='notice'>Using specialized micro tools you begin disconnecting the [target_move] from its frame....</span>")
+				if(do_after(user, 100, target = target_move))
+					user.remove_ventcrawl()
+					user.forceMove(target_move.loc) //handles entering and so on
+					user.visible_message("<span class='boldnotice'>You hear something squeezing through the ducts. With a resounding snap the [target_move] is fastened back in place.</span>", \
+						"<span class='notice'>You climb out the ventilation system. With a resounding snap the [target_move] is fastened back in place.</span>")
 		else if(target_move.can_crawl_through())
-			if(returnPipenet() != target_move.returnPipenet())
-				user.update_pipe_vision(target_move)
 			user.loc = target_move
 			user.client.eye = target_move //if we don't do this, Byond only updates the eye every tick - required for smooth movement
+			if(returnPipenet() != target_move.returnPipenet())
+				user.update_pipe_vision()
 			if(world.time - user.last_played_vent > VENT_SOUND_DELAY)
 				user.last_played_vent = world.time
 				playsound(src, 'sound/machines/ventcrawl.ogg', 50, 1, -3)
 	else
 		if((direction & initialize_directions) || is_type_in_list(src, GLOB.ventcrawl_machinery)) //if we move in a way the pipe can connect, but doesn't - or we're in a vent
-			user.remove_ventcrawl()
-			user.forceMove(src.loc)
-			user.visible_message("You hear something squeezing through the pipes.", "You climb out the ventilation system.")
-	user.canmove = 0
+			if(do_after(user, 30, target = src))
+				user.remove_ventcrawl()
+				user.forceMove(src.loc)
+				user.visible_message("<span class='boldnotice'>You hear something squeezing through the pipes.</span>", "<span class='notice'>You climb out the ventilation system.</span>")
+			else
+				to_chat(user, "<span class='notice'>You stop climbing out of the ventilation system.</span>")
+	user.canmove = FALSE
 	spawn(1)
-		user.canmove = 1
+		user.canmove = TRUE
 
 /obj/machinery/atmospherics/AltClick(var/mob/living/L)
-	if(is_type_in_list(src, GLOB.ventcrawl_machinery))
+	if(is_type_in_list(src, GLOB.ventcrawl_machinery) || check_open())
 		L.handle_ventcrawl(src)
 		return
 	..()
 
 /obj/machinery/atmospherics/proc/can_crawl_through()
-	return 1
+	return TRUE
+
+/obj/machinery/atmospherics/proc/check_open()
+	for(var/direction in GLOB.cardinal)
+		if(!findConnecting(direction) && direction & initialize_directions)
```

Comment:
```suggestion
		if((direction & initialize_directions) && !findConnecting(direction))
```
()s not required but makes it a bit more readable. The & is also cheaper to do so it's best to do that one first.
Can you add a comment here saying what it does btw? Atmos is quite ... interesting so to say so a comment here would help